plugins/localStyles.js

```diff
@@ -0,0 +1,351 @@
+
+      // Store the new (inlined + existing inline) CSS
+      // We don't stringify at match because this would be unnecessarily costly,
+      // when there are more selectors that match.
+      if(!matchedElement.newAstDeclarations)  matchedElement.newAstDeclarations = [];
+      matchedElement.newAstDeclarations = matchedElement.newAstDeclarations.concat(newAstDeclarations);
+
+      matchedElements.push(matchedElement);
+
+      return;
+    }
+
+    // Apply the new styles to their matched elements
+    // Global variables used: matchedElements
+    function reapplyElementStyles() {
+      var matchedElementsUniq  = uniq(matchedElements);
+      matchedElementsUniq.forEach(function(matchedElement) {
+        var newAstDeclarations = matchedElement.newAstDeclarations;
+        delete matchedElement.newAstDeclarations; // clean up XML AST
+
+        var newInlineCss = declarationsAstToInlineCss(newAstDeclarations);
+        // Note: We don't minifiy or deduplicate anything here,
+        // this is not the scope of this plugin.
+        // Use minifyStyles plugin after this one.
+        matchedElement.attr('style').value = newInlineCss;
+
+        return;
+      });
+
+      return;
+    }
+
+    // Is this CSS AST rule without any selectors (empty selectors)?
+    function isRuleWithoutSelectors(astRule) {
+      return !(astRule.selectors && astRule.selectors.length > 0);
+    }
+
+    // Apply the stripped down styles on the styles elements
+    // Global variables used: styleElements
+    function reapplyStyleStyles() {
+      styleElements.forEach(function(styleElement) {
+
+        var styleElementAst = styleElement.stylesAst;
+        delete styleElement.stylesAst; // clean up XML AST (2)
+
+        cleanupStyleRules(styleElementAst);
+
+        var newStyleCss = stringifyAst(styleElementAst);
+        styleElement.content[0].text = newStyleCss;
+      });
+      return;
+    }
+
+    // Clean up style rules
+    function cleanupStyleRules(styleElementAst) {
+      // clean up rules that were stripped down to no selectors at all
+      styleElementAst.stylesheet.rules.forEach(function(styleAstRule) {
+        if(!isTopRule(styleAstRule)) {
+          return; // skip
+        }
+
+        if(isRuleWithoutSelectors(styleAstRule)) {
+          removeValue(styleElementAst.stylesheet.rules, styleAstRule);
+        }
+        return;
+      });
+    }
+
+
+    // Recursion
+
+    // Default: Always select an item
+    function selectAll() {
+      return true;
+    }
+
+    /**
+     * Recursively traverses the HTML AST
+     *
+     * @param {Array}    items input items
+     * @param {Function} fn for processing an item
+     * @param {Function} fn for selecting an item
+     *
+     * @return {Array} output items
+     */
+    function monkeys(items, callFn, selectFn) {
+      if(selectFn  === undefined) { selectFn  = selectAll;     }
+
+        var      i = 0,
+            length = items.content.length;
+
+        while(i < length) {

```

Comment:
Why not just simple `for` loop?
```js
for (var i = 0; i < items.content.length; i++) {
```
Modern engines optimizes it, so there is no need in caching `length = items.content.length`.